packages/material-ui-utils/src/elementTypeAcceptingRef.js

```diff
@@ -0,0 +1,50 @@
+import * as PropTypes from 'prop-types';
+import React from 'react';
+import { isLazy, isMemo } from 'react-is';
+import chainPropTypes from './chainPropTypes';
+
+function isClassComponent(elementType) {
+  // elementType.prototype?.isReactComponent
+  const { prototype = {} } = elementType;
+
+  return Boolean(prototype.isReactComponent);
+}
+
+function elementTypeAcceptingRef(props, propName, componentName, location, propFullName) {
+  const propValue = props[propName];
+  const safePropName = propFullName || propName;
+
+  if (propValue == null) {
+    return null;
+  }
+
+  let warningHint;
+
+  /**
+   * blacklisting instead of whitelisting
+   *
+   * blacklisting will miss some components like React.Fragment. Those will at least
+   * trigger a warning in react.
+   * We can't whitelist because there is no safe way to detect React.forwardRef
+   * or class components. Safe means there's no public API.
```

Comment:
```suggestion
   * or class components. 'Safe' means there's no public API.
```

Reply:
Are you sure about the punctuation here? Shouldn't it be `"` instead of `'`?

Reply:
Fixed.